Failing on the reconciliation: " Failed to watch *v1.ConfigMap: unknown (get configmaps)`"

**Describe the bug**

I deployed the ArgoCD RBAC Operator using the Helm chart in local Minikube version 1.34.0. While the operator deploys successfully, when I try to apply the ArgoCDRole and ArgoCDBinding manifests results in the following error:

`E0108 22:55:14.417632       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.1/tools/cache/reflector.go:232: Failed to watch *v1.ConfigMap: unknown (get configmaps)`

This might not be a bug, but I'm trying to understand what I'm doing wrong here. 

Attaching the complete log for reference:

```
2025-01-08T22:54:56Z    INFO    setup    starting manager
2025-01-08T22:54:56Z    INFO    starting server    {"name": "health probe", "addr": ":8081"}
I0108 22:54:56.489824       1 leaderelection.go:250] attempting to acquire leader lease argocd/8becd400.my.domain...
I0108 22:55:14.305585       1 leaderelection.go:260] successfully acquired lease argocd/8becd400.my.domain
2025-01-08T22:55:14Z    DEBUG    events    argocd-controller-manager-66fbbf6845-tm7j8_adf2b708-8658-461e-8ac5-e70909685db4 became leader    {"type": "Normal", "object": {"kind":"Lease","namespace":"argocd","name":"8becd400.my.domain","uid":"2f82ee3a-18bd-413c-86bd-034882b53846","apiVersion":"coordination.k8s.io/v1","resourceVersion":"4510"}, "reason": "LeaderElection"}
2025-01-08T22:55:14Z    INFO    Starting EventSource    {"controller": "argocdrole", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRole", "source": "kind source: *v1alpha1.ArgoCDRole"}
2025-01-08T22:55:14Z    INFO    Starting EventSource    {"controller": "argocdrolebinding", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRoleBinding", "source": "kind source: *v1alpha1.ArgoCDRoleBinding"}
2025-01-08T22:55:14Z    INFO    Starting Controller    {"controller": "argocdrole", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRole"}
2025-01-08T22:55:14Z    INFO    Starting Controller    {"controller": "argocdrolebinding", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRoleBinding"}
2025-01-08T22:55:14Z    INFO    Starting workers    {"controller": "argocdrolebinding", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRoleBinding", "worker count": 1}
2025-01-08T22:55:14Z    INFO    Starting workers    {"controller": "argocdrole", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRole", "worker count": 1}
E0108 22:55:14.417632       1 reflector.go:150] pkg/mod/k8s.io/client-go@v0.30.1/tools/cache/reflector.go:232: Failed to watch *v1.ConfigMap: unknown (get configmaps)
2025-01-08T22:55:14Z    INFO    Observed a panic in reconciler: assignment to entry in nil map    {"controller": "argocdrole", "controllerGroup": "rbac-operator.argoproj-labs.io", "controllerKind": "ArgoCDRole", "ArgoCDRole": {"name":"test-role","namespace":"argocd"}, "namespace": "argocd", "name": "test-role", "reconcileID": "9cb91116-cb1d-495e-92cc-842e9cf70e48"}
panic: assignment to entry in nil map [recovered]
    panic: assignment to entry in nil map
```

**To Reproduce**

- Deploy the ArgoCD RBAC Operator using the Helm chart.
- Create the ArgoCDRole and ArgoCDBinding manifests in the same namespace (argocd).

**Expected behavior**

ConfigMap `argocd-rbac-cm` created after the ArgoCDRole and ArgoCDRoleBinding has been reconciled


**Additional context**

- The argocd namespace exists and the Crs are created in the same namespace
- Image tag v0.1.6

Thanks for the help!

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -13,14 +13,14 @@ jobs:
     timeout-minutes: 10
     steps:
       - name: Set up Go 1.x
-        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # pin@v4
+        uses: actions/setup-go@v4
         with:
           go-version: "1.21"
         id: go
       - name: Check out code into the Go module directory
-        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # pin@v3
+        uses: actions/checkout@v3
       - name: Restore Go build cache
-        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # pin@v3
+        uses: actions/cache@v3
         with:
           path: ~/.cache/go-build
           key: ${{ runner.os }}-go-build-v1-${{ github.run_id }}
@@ -35,13 +35,13 @@ jobs:
     env:
       GOPATH: /home/runner/go
     steps:
-      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # pin@v3
+      - uses: actions/checkout@v3
       - name: Setup Golang
-        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # pin@v4
+        uses: actions/setup-go@v4
         with:
           go-version: "1.21"
       - name: Restore Go build cache
-        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # pin@v3
+        uses: actions/cache@v3
         with:
           path: ~/.cache/go-build
           key: ${{ runner.os }}-go-build-v1-${{ github.run_id }}

--- a/config/rbac/role.yaml
+++ b/config/rbac/role.yaml
@@ -9,7 +9,9 @@ rules:
   resources:
   - configmaps
   verbs:
+  - get
   - list
+  - watch
 - apiGroups:
   - rbac-operator.argoproj-labs.io
   resources:

--- a/internal/controller/argocdrole_controller.go
+++ b/internal/controller/argocdrole_controller.go
@@ -46,7 +46,7 @@ type ArgoCDRoleReconciler struct {
 // +kubebuilder:rbac:groups=rbac-operator.argoproj-labs.io,resources=argocdroles/status,verbs=*
 // +kubebuilder:rbac:groups=rbac-operator.argoproj-labs.io,resources=argocdroles/finalizers,verbs=*
 // +kubebuilder:rbac:groups=rbac-operator.argoproj-labs.io,resources=argocdrolebindings,verbs=get;list
-// +kubebuilder:rbac:groups="",resources=configmaps,verbs=list
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.

--- a/internal/controller/argocdrolebinding_controller.go
+++ b/internal/controller/argocdrolebinding_controller.go
@@ -44,7 +44,7 @@ type ArgoCDRoleBindingReconciler struct {
 // +kubebuilder:rbac:groups=rbac-operator.argoproj-labs.io,resources=argocdrolebindings/finalizers,verbs=*
 // +kubebuilder:rbac:groups=rbac-operator.argoproj-labs.io,resources=argocdroles,verbs=get;list
 // +kubebuilder:rbac:groups=rbac-operator.argoproj-labs.io,resources=argocdroles/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups="",resources=configmaps,verbs=list
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.

--- a/internal/controller/configmap.go
+++ b/internal/controller/configmap.go
@@ -93,6 +93,10 @@ func (r *ArgoCDRoleReconciler) reconcileRBACConfigMap(cm *corev1.ConfigMap, role
 	overlayKey := fmt.Sprintf("policy.%s.%s.csv", role.Namespace, role.Name)
 	roleName := fmt.Sprintf("role:%s", role.Name)
 
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
+	}
+
 	// Default Policy String
 	if cm.Data[common.ArgoCDKeyRBACPolicyCSV] != getDefaultRBACPolicy() {
 		cm.Data[common.ArgoCDKeyRBACPolicyCSV] = getDefaultRBACPolicy()
@@ -166,6 +170,10 @@ func (r *ArgoCDRoleBindingReconciler) reconcileRBACConfigMap(cm *corev1.ConfigMa
 func (r *ArgoCDRoleBindingReconciler) reconcileRBACConfigMapForBuiltInRole(cm *corev1.ConfigMap, rb *rbacoperatorv1alpha1.ArgoCDRoleBinding, role *rbacoperatorv1alpha1.ArgoCDRole) error {
 	changed := false
 	overlayKey := fmt.Sprintf("policy.%s.%s.csv", role.Namespace, role.Name)
+
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
+	}
 
 	// Default Policy String
 	if cm.Data[common.ArgoCDKeyRBACPolicyCSV] != getDefaultRBACPolicy() {

--- a/internal/controller/configmap.go
+++ b/internal/controller/configmap.go
@@ -115,6 +115,10 @@ func (r *ArgoCDRoleReconciler) reconcileRBACConfigMapWithRoleBinding(cm *corev1.
 	changed := false
 	overlayKey := fmt.Sprintf("policy.%s.%s.csv", role.Namespace, role.Name)
 
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
+	}
+
 	// Default Policy String
 	if cm.Data[common.ArgoCDKeyRBACPolicyCSV] != getDefaultRBACPolicy() {
 		cm.Data[common.ArgoCDKeyRBACPolicyCSV] = getDefaultRBACPolicy()
@@ -136,6 +140,10 @@ func (r *ArgoCDRoleReconciler) reconcileRBACConfigMapWithRoleBinding(cm *corev1.
 func (r *ArgoCDRoleBindingReconciler) reconcileRBACConfigMap(cm *corev1.ConfigMap, rb *rbacoperatorv1alpha1.ArgoCDRoleBinding, role *rbacoperatorv1alpha1.ArgoCDRole) error {
 	changed := false
 	overlayKey := fmt.Sprintf("policy.%s.%s.csv", role.Namespace, role.Name)
+
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
+	}
 
 	// Default Policy String
 	if cm.Data[common.ArgoCDKeyRBACPolicyCSV] != getDefaultRBACPolicy() {

